tests/mocks/message_bus.py

```diff
@@ -70,23 +73,28 @@ def consumer(self, partition_id, type):
 class SpiderFeedStream(BaseSpiderFeedStream):
 
     def __init__(self, messagebus):
-        self.ready_partitions = set(messagebus.spider_feed_partitions)
+        self._producer = Producer()
+        self.max_next_requests = messagebus.max_next_requests
+        self.partitions_offset = {}
+        for partition_id in messagebus.spider_feed_partitions:
+            self.partitions_offset[partition_id] = 0
 
     def producer(self):
-        return Producer()
+        return self._producer
 
     def consumer(self, partition_id):
         return Consumer()
 
     def available_partitions(self):
-        return self.ready_partitions
-
-    def mark_ready(self, partition_id):
-        self.ready_partitions.add(partition_id)
-
-    def mark_busy(self, partition_id):
-        self.ready_partitions.discard(partition_id)
-
+        partitions = []
+        for partition_id, last_offset in self.partitions_offset.items():
```

Comment:
Here, if partition doesn't exist (yet) in this dict - it will not be returned as available, which is wrong. 
What if producer offsets will be less (because of DB worker restart) than consumer offsets?